### PR TITLE
Add GitHub Action for AI PR Summarizer

### PR DESCRIPTION
**Pull request by sberney**
This PR uses a [Pull Request Summarizer](https://github.com/marketplace/actions/pull-request-summarizer) action to automatically summarize PRs with GPT.

I checked out the repo and ran `npx add-gpt-summarizer@latest`, committed the results, and created this PR.

In order for it to function, you need to follow [its directions](https://github.com/marketplace/actions/pull-request-summarizer#setting-up) to create a repository secret for your OpenAI API Key.

> ... To do this, go to your repository's settings and navigate to the "Secrets" section. Click on "Add a new secret" and enter the secret name OPENAI_API_KEY and the value of your API key.

### CODE REVIEW

File: .github/workflows/gpt-commit-summarizer.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# summaries, and delete outdated code review comments
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  summarize:
+    runs-on: ubuntu-latest
+    permissions: write-all  # Some repositories need this line
```
Comment (Torantulino) [inline .github/workflows/gpt-commit-summarizer.yml]: This seems pretty sketchy...
Anyone want to weigh in?

Comment (Torantulino) [reply .github/workflows/gpt-commit-summarizer.yml]: Specifically talking about it needing write-all permissions.

Comment (sberney) [reply .github/workflows/gpt-commit-summarizer.yml]: Good catch. I'll update the PR later today.

**Comment by sberney** [comment]
It looks like it just needs the `pull-requests: write` permission, but I got tripped up since during testing I tried to initiate a PR from your repo into my fork, and it kept giving me some kind of permission error. But once I tried to create a PR from my repo into my own repo, it started working fine again. I don't know if that's something that would be seen in the real world.

Example of it functioning:
https://github.com/sberney/Auto-GPT/pull/10

**Comment by ntindle** [comment]
@sberney give this an update from master

**Comment by ntindle** [comment]
I appreciate this, but our workflows have changed significantly since this was introduced. I'm going to go ahead and close this for now to help clear our backlog of PRs

**Pull request closed without merging**